blog: July 2025 Security Releases (#1994)
expressjs/expressjs.com: _posts/2025-07-31-security-releases.md

---
title: July 2025 Security Releases
description: Security releases for Multer and On-headers has been published. We recommend that all users upgrade as soon as possible.
tags: security vulnerabilities
authors:
  - name: Ulises Gascón
    github: UlisesGascon
---

> The Express team has released a new path version of [Multer](https://www.npmjs.com/package/multer) addressing a high-severity security vulnerability. Also a new minor version of [on-headers](https://www.npmjs.com/package/on-headers) addressing a low-severity security vulnerability

Reviewer: ```suggestion
The Express team has released a new patch version of [Multer](https://www.npmjs.com/package/multer) addressing a high-severity security vulnerability. Also a new minor version of [on-headers](https://www.npmjs.com/package/on-headers) addressing a low-severity security vulnerability
```

> The Express team has released a new path version of [Multer](https://www.npmjs.com/package/multer) addressing a high-severity security vulnerability. Also a new minor version of [on-headers](https://www.npmjs.com/package/on-headers) addressing a low-severity security vulnerability

Reviewer 2: ```suggestion
The Express team has released a new patch version of [Multer](https://www.npmjs.com/package/multer) addressing a high-severity security vulnerability, and a new minor version of [on-headers](https://www.npmjs.com/package/on-headers) addressing a low-severity security vulnerability.
```

Author: Solved in [`4ee2d89` (#1994)](https://github.com/expressjs/expressjs.com/pull/1994/commits/4ee2d89c485ead8b7e75b91767367bde5d41e0d8)

> The Express team has released a new patch version of [Multer](https://www.npmjs.com/package/multer) addressing a high-severity security vulnerability. Also a new minor version of [on-headers](https://www.npmjs.com/package/on-headers) addressing a low-severity security vulnerability

Author: ```suggestion
The Express team has released a new patch version of [Multer](https://www.npmjs.com/package/multer) addressing a high-severity security vulnerability, and a new minor version of [on-headers](https://www.npmjs.com/package/on-headers) addressing a low-severity security vulnerability.
```

Author: mimic https://github.com/expressjs/expressjs.com/pull/1994#discussion_r2245500994

{% include admonitions/warning.html 
content="We recommend upgrading to the latest version of Multer and On-headers immediately to secure your applications."
%}

The following vulnerabilities have been addressed:

- [High severity vulnerability CVE-2025-7338 in Multer middleware](#high-severity-vulnerability-cve-2025-7338-in-multer-middleware)
- [Low severity vulnerability CVE-2025-7339 in On-header middleware](#low-severity-vulnerability-cve-2025-7339-in-on-header-middleware)

## High severity vulnerability CVE-2025-7338 in Multer middleware

> **[Multer](https://www.npmjs.com/package/multer) versions `>=1.4.4-lts.1` and `<2.0.2` are vulnerable to denial of service ia unhandled exception from malformed request.**

Reviewer 3: `ia` -> `via`

```suggestion
**[Multer](https://www.npmjs.com/package/multer) versions `>=1.4.4-lts.1` and `<2.0.2` are vulnerable to denial of service via unhandled exception from malformed request.**
```

This request causes an unhandled exception, leading to a crash of the process.

**Affected versions**: `>=1.4.4-lts.1, <2.0.2`  

> **Patched version**: `>=2.0.2`

Reviewer 2: ```suggestion
**Patched version**: `2.0.2`
```

For more details, see [GHSA-fjgf-rc76-4x9p](https://github.com/expressjs/multer/security/advisories/GHSA-fjgf-rc76-4x9p).

## Low severity vulnerability CVE-2025-7339 in On-header middleware

**[On-headers](https://www.npmjs.com/package/on-headers) versions `<1.1.0` is vulnerable to http response header manipulation**

A bug in on-headers versions `<1.1.0` may result in response headers being inadvertently modified when an array is passed to `response.writeHead()`

**Affected versions**: `<1.1.0`  

> **Patched version**: `>=1.1.0`

Reviewer 2: ```suggestion
**Patched version**: `1.1.0`
```
